Competition2017/src/competition/subsystems/drive/DriveSubsystem.java

```diff
@@ -5,37 +5,160 @@
 
-        this.leftFrontDrive = factory.getSpeedController(0);
-        this.leftRearDrive = factory.getSpeedController(2);
+        // TODO: Update these defaults. The current values are blind guesses.
+        encoderCodesProperty = propManager.createPersistentProperty("Drive encoder codes per rev", 512);
+        maxSpeedProperty = propManager.createPersistentProperty("Max drive motor speed (rotations per second)", 5);
 
-        this.rightFrontDrive = factory.getSpeedController(1);
-        this.rightFrontDrive.setInverted(true);
-        this.rightRearDrive = factory.getSpeedController(3);
-        this.rightRearDrive.setInverted(true);
+        p = propManager.createPersistentProperty("Drive P", 2);
+        i = propManager.createPersistentProperty("Drive I", 0);
+        d = propManager.createPersistentProperty("Drive D", -100);
+        f = propManager.createPersistentProperty("Drive F", 0);
+        
+        this.leftDrive = factory.getCANTalonSpeedController(3);
+        this.leftDriveSlave = factory.getCANTalonSpeedController(4);
+        configMotorTeam(leftDrive, leftDriveSlave);
+        leftDrive.createTelemetryProperties("Left master", propManager);
+        leftDriveSlave.createTelemetryProperties("Left slave", propManager);
+        
+        this.rightDrive = factory.getCANTalonSpeedController(1);
+        this.rightDriveSlave = factory.getCANTalonSpeedController(2);
+        configMotorTeam(rightDrive, rightDriveSlave);
+        rightDrive.createTelemetryProperties("Right master", propManager);
+        rightDriveSlave.createTelemetryProperties("Right slave", propManager);
     }
 
-    public void tankDrive(double leftPower, double rightPower) {
-        this.leftFrontDrive.set(leftPower);
-        this.leftRearDrive.set(leftPower);
-        this.rightFrontDrive.set(rightPower);
-        this.rightRearDrive.set(rightPower);
+    private void configMotorTeam(XCANTalon master, XCANTalon slave) {
+        // TODO: Check faults and voltage/temp/current
+        
+        // Master config
+        master.setFeedbackDevice(FeedbackDevice.QuadEncoder);
+        master.setBrakeEnableDuringNeutral(false);
+        master.reverseSensor(false);
+        master.enableLimitSwitches(false, false);
+        
+        master.configNominalOutputVoltage(0,  -0);
+        master.configPeakOutputVoltage(12, -12);
+
+        master.setProfile(0);
+        master.setF(0);
+        updateMotorConfig(master);
+        master.setControlMode(TalonControlMode.Speed);
+        //master.setControlMode(TalonControlMode.PercentVbus);
+        
+        master.set(0);
+        
+        // Slave config
+        slave.configNominalOutputVoltage(0,  -0);
+        slave.configPeakOutputVoltage(12, -12);
+        slave.enableLimitSwitches(false, false);
+        
+        slave.setControlMode(TalonControlMode.Follower);
+        slave.set(master.getDeviceID());
+    }
+    
+    public double convertPowerToVelocityTarget(double power) {
```

Comment:
since there was some confusion over what this method does/is used for, might be worth a quick method comment